### DISCUSSION TITLE
Lesson6: FoundrySimpleStorage - getting segmentation fault while compiling using ```forge create SimpleStorage --interactive```

### DISCUSSION THREAD

**Discussion by elec7rik** [Cyfrin/foundry-full-course-cu#758] [Q&A] [answered]
I was trying to compile the SimpleStorage on anvil using the command:
```forge create SimpleStorage --interactive
```
On the right pane, after pasting the key it was getting stuck, error said there was an invalid character u// in the key(there was actually none). Then the left anvil window was getting ```core dumped, segmentation fault```.  I checked on phind and there was a SO answer to a seemingly similar problem in which an answer had suggested to change the key to UTF-8 and then do it, it didn't work either. 
Now I'm not even getting up to the password pasting point due to this error. What should I do?

![Screenshot from 2023-08-30 17-05-33](https://github.com/Cyfrin/foundry-full-course-f23/assets/89316185/6a6f2e65-e249-4955-b9a6-34c9d49fcd4f)

**Comment by theirrationalone** [comment]
I recently made `foundryup` into my bash... Now i'm also getting this fault. There might be bugs into new update. but scripts are working fine  without broadcast.

**Reply by elec7rik** [reply]
thanks, but how can i deploy to anvil without this working? also, what do you mean by "scripts are working fine without broadcast."

**Reply by theirrationalone** [reply, marked as answer]
You have anvil running in a separated bash terminal, Now on a new bash terminal when you would try to run scripts i.e.,`forge script script/script_name.s.sol:contract_name --rpc-url <url> --private-key <private_key>`, are working well but for on chain simulation same script command appended with `--broadcast`, aren't working at all.

**Reply by theirrationalone** [reply]
You can rely on temporary anvil spin by foundry itself until fix. Means just execute your scripts without spinning anvil by on your own.

**Reply by elec7rik** [reply]
got it working, had to reinstall foundry. ```forge --version``` as well as other forge commands were also throwing seg fault meanwhile cast, anvil and chisel were working fine. thank you. commands with ```--broadcast``` also seem to be working fine now. i just hope this doesn't lead to an issue where i have to install foundry every time before using forge,

**Reply by elec7rik** [reply]
ahan there seems to be some old issue, even mentioned in the foundry book.
![forge](https://github.com/Cyfrin/foundry-full-course-f23/assets/89316185/eb5d8869-7b1b-467e-8f5c-7348cf2abd2c)
